pinot-core/src/test/java/org/apache/pinot/queries/MultiValueRawQueriesTest.java

```diff
@@ -338,6 +338,445 @@ public void testSelectQueries() {
+        int[] intVal = (int[]) values[0];
+        assertEquals(intVal[1] - intVal[0], MV_OFFSET);
+        assertEquals(intVal[0], i % 5);
+        assertEquals(intVal[1], (i % 5) + MV_OFFSET);
+
+        double[] doubleVal = (double[]) values[1];
+        assertEquals(doubleVal[0], (double) i % 5);
+        assertEquals(doubleVal[1], (double) (i % 5) + MV_OFFSET);
+
+        String[] stringVal = (String[]) values[2];
+        assertEquals(Integer.parseInt(stringVal[0]), i % 5);
+        assertEquals(Integer.parseInt(stringVal[1]), (i % 5) + MV_OFFSET);
+      }
+    }
+    {
+      // Test a select with filter query (OR) on two MV raw column identifiers (int and double)
+      String query = "SELECT mvRawIntCol, mvRawDoubleCol, mvRawStringCol from testTable where mvRawIntCol < 5 "
+          + "OR mvRawDoubleCol > 1104.0 LIMIT 10";
+      ResultTable resultTable = getBrokerResponse(query).getResultTable();
+      assertNotNull(resultTable);
+      DataSchema dataSchema = new DataSchema(new String[]{
+          "mvRawIntCol", "mvRawDoubleCol", "mvRawStringCol"
+      }, new DataSchema.ColumnDataType[]{
+          DataSchema.ColumnDataType.INT_ARRAY, DataSchema.ColumnDataType.DOUBLE_ARRAY,
+          DataSchema.ColumnDataType.STRING_ARRAY
+      });
+      assertEquals(resultTable.getDataSchema(), dataSchema);
+      List<Object[]> recordRows = resultTable.getRows();
+      assertEquals(recordRows.size(), 10);
+
+      for (int i = 0; i < 10; i++) {
+        Object[] values = recordRows.get(i);
+        assertEquals(values.length, 3);
+        int[] intVal = (int[]) values[0];
+        assertEquals(intVal[1] - intVal[0], MV_OFFSET);
+        double[] doubleVal = (double[]) values[1];
+        assertTrue(intVal[0] < 5 || intVal[1] < 5 || doubleVal[0] > 1104.0 || doubleVal[1] > 1104.0);
+      }
+    }
+    {
+      // Test a select with filter query on a long MV raw column identifier
+      String query = "SELECT mvRawLongCol from testTable where mvRawLongCol > 1100 LIMIT 10";
+      ResultTable resultTable = getBrokerResponse(query).getResultTable();
+      assertNotNull(resultTable);
+      DataSchema dataSchema = new DataSchema(new String[]{
+          "mvRawLongCol"
+      }, new DataSchema.ColumnDataType[]{
+          DataSchema.ColumnDataType.LONG_ARRAY
+      });
+      assertEquals(resultTable.getDataSchema(), dataSchema);
+      List<Object[]> recordRows = resultTable.getRows();
+      assertEquals(recordRows.size(), 10);
+
+      for (int i = 0; i < 10; i++) {
+        Object[] values = recordRows.get(i);
+        assertEquals(values.length, 1);
+        long[] longVal = (long[]) values[0];
+        assertEquals(longVal[1] - longVal[0], MV_OFFSET);
+        assertTrue(longVal[0] > 1100 || longVal[1] > 1100);
+      }
+    }
+    {
+      // Test a select with filter = query on a string MV raw column identifier
+      String query = "SELECT mvRawStringCol from testTable where mvRawStringCol = '1100' LIMIT 10";
```

Comment:
Didn't notice this in the part 1 PR. May be it is better to use some actual varying width string / alphanumeric in the MV String column just to ensure correctness ?

Reply:
As discussed offline we'll add a future PR in which we have alpha-numeric string MV raw column and add tests for this. Here we do have some varying length string values (len 1, 2, 3, 4).